__test__/contracts/Protocol/MemoryUtils/read.spec.ts

```diff
@@ -0,0 +1,78 @@
+import { viem } from 'hardhat';
+import { keccak256, toBytes, toHex } from 'viem/utils';
+
+async function deployTestMemoryUtils() {
+  const testMemoryUtilsContract = await viem.deployContract('TestMemoryUtils', []);
+
+  return {
+    testMemoryUtilsContract
+  };
+}
+
+describe('returns', function () {
+  let testMemoryUtils: Awaited<ReturnType<typeof deployTestMemoryUtils>>['testMemoryUtilsContract'];
+
+  beforeAll(async () => {
+    testMemoryUtils = (await deployTestMemoryUtils()).testMemoryUtilsContract;
+  });
```

Comment:
This test suite is critical to ensure future updates do not change memory locations.

It ensures we don't lock ourselves out of an admin slot for example